main.rkt

```diff
@@ -0,0 +1,218 @@
+  (string->url (format "~a/repos/~a/pulls/~a/reviews"
+                       (github-api-url)
+                       (github-repository)
+                       (github-review-request-pull-number req))))
+
+(define (github-review-request-send req)
+  (parameterize ([current-https-protocol 'secure])
+    (define response-port
+      (post-pure-port (github-review-request-url req)
+                      (jsexpr->bytes (github-review-request-body-jsexpr req))
+                      ; https://docs.github.com/en/rest/reference/pulls#list-review-comments-in-a-repository-preview-notices
+                      `("Accept: application/vnd.github.comfort-fade-preview+json"
+                        ,(format "Authorization: Bearer ~a" (github-token)))))
+    (define response (port->string response-port))
+    (close-input-port response-port)
+    response))
+
+; thank you jack for this code in the resyntax cli.rkt :)
+(define/guard (string-indent s #:amount amount)
+  (guard (zero? amount) then s)
+  (define indent-string (make-string amount #\space))
+  (define lines
+    (for/list ([line (in-lines (open-input-string s))])
+      (string-append indent-string line)))
+  (string-join lines "\n"))
+
+(define (resyntax-github-run)
+  (define filenames (git-diff-names (github-base-ref)))
+  (define files (file-groups-resolve (map single-file-group filenames)))
+  (printf "resyntax: --- analyzing code ---\n")
+  (define results
+    (transduce files
+               (append-mapping (refactor-file _ #:suite default-recommendations))
+               #:into into-list))
+  
+  (define comments
+    (for/list ([result (in-list results)])
+      (define path (file-source-path (refactoring-result-source result)))
+      (define old-code-snippet (refactoring-result-original-code result))
+      (define new-code-snippet (refactoring-result-new-code result))
+      (define start-line (code-snippet-start-line old-code-snippet))
+      (define end-line (sub1 (code-snippet-end-line old-code-snippet)))
+      (define start-col (code-snippet-start-column new-code-snippet))
+      (define new-code (code-snippet-raw-text new-code-snippet))
+      (define body (format "```suggestion\n~a\n```\n\nRule: `~a`\n~a"
+                           (string-indent new-code #:amount start-col)
+                           (refactoring-result-rule-name result)
+                           (refactoring-result-message result)))
+      (github-review-comment #:path (first (git-path path))
+                             #:body body
+                             #:start-line start-line
+                             #:end-line end-line
+                             #:start-side "RIGHT"
+                             #:end-side "RIGHT")))
+  
+  (define req
+    (github-review-request #:owner-repo (github-repository)
+                           #:pull-number (git-ref->pr-number (github-ref))
+                           #:commit-id (github-sha) 
+                           #:body (if (null? comments)
```

Comment:
For the messages, I think it should explicitly mention the name Resyntax, since otherwise it's kind of hard to tell in the comments what tool is producing these suggestions. So how about these messages:

- No comments: "Resyntax analyzed this pull request and found no issues."
- Some comments: "Resyntax analyzed this pull request and has suggestions."